devops/pipeline/binary_package.yaml

```diff
@@ -15,191 +15,232 @@ trigger:
+      displayName: Build & Stage Source Package
+
+    - publish: $(Build.ArtifactStagingDirectory)
+      displayName: Publishing source package
+      artifact: OSConfig_$(Build.BuildNumber)_SOURCES
+
+- stage: binary_package
+  displayName: Binary Package Build
+  dependsOn: []
+  jobs:
+  - job: build
+
+    strategy:
+      matrix:
+        Code Coverage:
+          variant: ubuntu20.04
+          arch: amd64_debug
+          containerImage: ubuntu20.04-dev-amd64:latest
+          platform: linux/amd64
+          coverage: true
+          ignoreTestFailures: false
+        Packaging ubuntu2004-amd64:
+          variant: ubuntu20.04
+          arch: amd64
+          containerImage: ubuntu20.04-dev-amd64:latest
+          platform: linux/amd64
+          coverage: false
+          ignoreTestFailures: false
+        Packaging ubuntu2004-arm:
+          variant: ubuntu20.04
+          arch: arm
+          containerImage: ubuntu20.04-dev-arm:latest
+          platform: linux/arm/v7
+          coverage: false
+          ignoreTestFailures: true
+        Packaging ubuntu2004-arm64:
+          variant: ubuntu20.04
+          arch: arm64
+          containerImage: ubuntu20.04-dev-arm64:latest
+          platform: linux/arm64/v8
+          coverage: false
+          ignoreTestFailures: false
+        Packaging ubuntu1804-amd64:
+          variant: ubuntu18.04
+          arch: amd64
+          containerImage: ubuntu18.04-dev-amd64:latest
+          platform: linux/amd64
+          coverage: false
+          ignoreTestFailures: false
+        Packaging ubuntu1804-arm:
+          variant: ubuntu18.04
+          arch: arm
+          containerImage: ubuntu18.04-dev-arm:latest
+          platform: linux/arm/v7
+          coverage: false
+          ignoreTestFailures: false
+        Packaging ubuntu18-arm64:
+          variant: ubuntu18.04
+          arch: arm64
+          containerImage: ubuntu18.04-dev-arm64:latest
+          platform: linux/arm64/v8
+          coverage: false
+          ignoreTestFailures: false
+        Packaging debian9-amd64:
+          variant: debian9
+          arch: amd64
+          containerImage: debian9-dev-amd64:latest
+          platform: linux/amd64
+          coverage: false
+          ignoreTestFailures: false
+        Packaging debian9-arm:
+          variant: debian9
+          arch: arm
+          containerImage: debian9-dev-arm:latest
+          platform: linux/arm/v7
+          coverage: false
+          ignoreTestFailures: false
+        Packaging debian9-arm64:
+          variant: debian9
+          arch: arm64
+          containerImage: debian9-dev-arm64:latest
+          platform: linux/arm64/v8
+          coverage: false
+          ignoreTestFailures: false
+
+    pool:
+      vmImage: 'ubuntu-20.04'
+
+    steps:
+
+    - checkout: self
+      submodules: recursive
+      clean: true
```

Comment:
From here to end of file - are these just formatting changes? And if so, why are they necessary? Blocks appear the same with exception of being spaced to the right (why?)

Reply:
yes, the new stuff is from 19-52 i added a new `source` stage which produces and publishes the source package to the pipeline

Reply:
That part is fine. Question is about the changes from here on line 136 to end of file - are these really necessary?

Reply:
yes, they need to be indented as we've introduced a more steps though